cristin-project/src/main/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandler.java

```diff
@@ -87,82 +50,28 @@ protected QueryCristinProjectHandler(QueryCristinProjectApiClient cristinApiClie
     protected SearchResponse<NvaProject> processInput(Void input, RequestInfo requestInfo, Context context)
             throws ApiGatewayException {
 
-        validateQueryParameterKeys(requestInfo);
-
-        var requestQueryParameters = extractQueryParameters(requestInfo);
-        if (requestInfo.getQueryParameters().containsKey(ORGANIZATION)) {
-            requestQueryParameters.put(ORGANIZATION, getValidOrganizationUri(requestInfo));
-        }
-        getValidProjectStatus(requestInfo).ifPresent(status -> requestQueryParameters
-                .put(STATUS, status.name()));
-        return getTransformedCristinProjectsUsingWrapperObject(requestQueryParameters);
+        final var cristinQuery =
+            CristinQuery.builder()
+                .fromRequestInfo(requestInfo)
+                .withRequiredParameters(REQUIRED_QUERY_PARAMETER)
+                .asNvaQuery()
+                .validate()
+                .build();
+        return cristinApiClient.queryCristinProjectsIntoWrapperObjectWithAdditionalMetadata(cristinQuery);
 
 
     }
 
-    private Map<String, String> extractQueryParameters(RequestInfo requestInfo) throws BadRequestException {
-        Map<String, String> requestQueryParameters = new ConcurrentHashMap<>();
-        requestQueryParameters.put(LANGUAGE, getValidLanguage(requestInfo));
-        requestQueryParameters.put(QUERY, getValidQuery(requestInfo));
-        requestQueryParameters.put(PAGE, getValidPage(requestInfo));
-        requestQueryParameters.put(NUMBER_OF_RESULTS, getValidNumberOfResults(requestInfo));
```

Comment:
Some of these used to have default values. We should consider keeping it that way

Reply:
They still have default values. and are set, if they are required

I tried to make the code generic, so it could be sub classed and used everywhere.